lib/src/git_backend.rs

```diff
@@ -159,18 +161,24 @@ pub struct GitBackend {
     empty_tree_id: TreeId,
     extra_metadata_store: TableStore,
     cached_extra_metadata: Mutex<Option<Arc<ReadonlyTable>>>,
+    change_id_setting: bool,
 }
 
 impl GitBackend {
     pub fn name() -> &'static str {
         "git"
     }
 
-    fn new(base_repo: gix::ThreadSafeRepository, extra_metadata_store: TableStore) -> Self {
+    fn new(
+        base_repo: gix::ThreadSafeRepository,
+        extra_metadata_store: TableStore,
+        settings: &UserSettings,
```

Comment:
nit: I think I would prefer to pass only the boolean value here instead since this is pretty low-level code and it needs only that value.